### PR TITLE
feat: create basic meson build script

### PR DESCRIPTION
**Pull request by Sophon96**

### CODE REVIEW

**Review by JakeRoggenbuck — APPROVED** [review]
Tested, lgtm

**Comment by adamhutchings** [comment]
What operating systems has this been tested on?

File: core/meson.build [hunk 1]
```diff
@@ -0,0 +1,6 @@
+core_sources = ['src/math/matrix.cpp', 'src/math/vector.cpp']
```
Comment (adamhutchings) [inline core/meson.build:1]: Are we going to have to update this line every time we add a new `.cpp` file to the library? Do you know if there's a way to do this automatically?

**Comment by Sophon96** [comment]
Why the heck won't this darn workflow run.

**Comment by Sophon96** [comment]
> What operating systems has this been tested on?

Windows and Linux (I assume that's what Jake is using).

**Comment by JakeRoggenbuck** [comment]
> > What operating systems has this been tested on?
> 
> Windows and Linux (I assume that's what Jake is using).

yup! It builds an output .so that looks good

**Comment by Sophon96** [comment]
Awesome! Any notes on the docs (just `README.md` for now)?

**Review by adamhutchings — CHANGES_REQUESTED** [review]
Unfortunately, this doesn't work on macOS.

```
../core/src/math/matrix.cpp:39:40: error: no member named 'make_unique' in namespace 'std'
    std::unique_ptr<Vector> ret = std::make_unique<Vector>(this->m);
                                  ~~~~~^
../core/src/math/matrix.cpp:39:52: error: 'Vector' does not refer to a value
    std::unique_ptr<Vector> ret = std::make_unique<Vector>(this->m);
```

This is what happens when compiling. Luckily, I'm pretty sure the only fix needed is to set the build version to C++14.

File: core/meson.build (hunk 1 above)
Comment (Sophon96) [reply core/meson.build:1]: Yes, that's the preferred pattern. It's explained here: https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard

**Comment by Sophon96** [comment]
I am literally going to throw my mouse out of the window. Why won't this workflow run?!

**Comment by Sophon96** [comment]
@adamhutchings Test again?

**Review by adamhutchings — APPROVED** [review]

**Comment by adamhutchings** [comment]
Are we good to merge?

**Comment by Sophon96** [comment]
Should be good!

> Are we good to merge?

**Comment by Sophon96** [comment]
I'll fix the (undoubtedly broken) workflow once this is merged.

**Comment by Sophon96** [comment]
I await the incredibly painful toolchain teething issues ahead!

**Pull request merged**